docs/interview/06-ExpressJS/01-easy/01-easy.md

```diff
@@ -21,15 +21,79 @@ serve data to client (React) applications.
 
 <details>
   <summary>What is API?</summary>
-  TODO: add answer @PournimaAnjole
+
+**APIs:** API stands for "Application Programming Interface", is a set of rules and tools that allows different software applications to communicate with each other.
+An API is like a messenger between different computer programs. It helps them share information and work together smoothly.
+
 </details>
 
 <details>
   <summary>What are the various HTTP methods?</summary>
-  TODO: add answer @PournimaAnjole
+
+**HTTP Methods:** HTTP methods (Hypertext Transfer Protocol methods).
+HTTP methods provide a way for clients to interact with web servers in a standardized manner. Each method corresponds to a specific action that the server should take regarding the requested resource. For example, a GET request is used when you want to retrieve data, and a POST request is used when you want to send data to be processed or stored.The most common HTTP methods are:
+
+**1.GET:** Used to request data from a specified resource.
+
+**2.POST:** Used to submit data to be processed to a specified resource. Often used for submitting form data or uploading a file.
+
+**3.PUT:** Used to update a resource or create a new resource if it does not exist at the specified URI.
+
+**4.DELETE:** Used to request the removal of a resource.
+
+**5.PATCH:** Used to apply partial modifications to a resource.
+
 </details>
 
 <details>
   <summary>What is middleware?</summary>
-  TODO: add answer @PournimaAnjole
+
+**middleware:** middleware is like a communication helper, translator, and security guard for computer systems, ensuring they work together smoothly and safely.
+Middleware often includes security features to protect data during communication. This can involve encryption, authentication, and authorization mechanisms.
+
+middleware functions often receive three parameters: req (request), res (response), and next.
+
+**1.req (request):** This parameter contains information about the incoming HTTP request, including data like headers, parameters, and the request body.
+
+**2.res (response):** This parameter is used to construct and send the HTTP response back to the client.
+
+**3.next:** This is a function that, when called, passes control to the next middleware in the chain. If next is not called within a middleware function, the request-response cycle may be terminated, and the response may be sent back to the client.
+
+```js showLineNumbers=true
+const validprasam = (req, res, next) => {
```

Comment:
`validprasam` typo is there change it to `validparams`

`validateParams` is the appropriate name